Modify Contribution file (#13)
redhat-beyond/beyond08-group3: CONTRIBUTING.md

# CONTRIBUTING
Contributing to the official buyTogether application
Welcome! We're so excited that you're considering contributing to the official GitHub page for buyTogether.
---
## Prerequisites
- Fork the repo to make your changes without affecting the original project until you're ready to merge.
- Create a working branch.
- Set up a development environment to begin working on the project, all you need to do is 'vagrant up'
- Start with your changes!
---
## Commiting
Please use the guide below inorder to follow our commit message convention:

> * [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)

Reviewer: This very commit does not follow the guide that you reference. e.g. the commit message title does not follow that format.

Also, I don't want the body to be optional. It should be mandatory.

Author: you're right, i didnt notice those details.

---
## Pull Request
- When you're finished with the changes, create a pull request, also known as a PR.
- If your PR depends on the merging of other PRs, indicate this on the commit message or on the comments.
Specifically, if it contains unmerged commits of any other PRs,
indicate to the reviewers which commits are expected to be reviewed as part of your PR.
- Keep your PRs as small as possible.
Ideally, a single PR should include a single commit.
- The reviewer resolves the conversation to indicate that they are ok with the updated content.
- If you run into any merge issues, check how to solve them and resolve merge conflicts and other issues.
- Changes made as part of the review process (i.e. fixes to the content of the original commit) ,

> should be added to the original commit (e.g. using ```commit --amend```), rather than creating a new commit. 

Reviewer: ```suggestion
should be added to the original commit (e.g. using `commit --amend`), rather than creating a new commit. 
```
You should use this triple thingy when you're adding a code block. When adding inline code, you just do a single-thingy.

```
the block
is hot
```

vs. `the fox is black`.

Author: commit --amend doesnt count as code block?

Reviewer: If it's on a line of its own (or multiple lines), you use \`\`\`. If it's together on the same line with regular text you use a single \`

- Please keep in mind that at least two team members of the group should "quality review"(i.e. NO LGTM) and approve the pull request before asking the mentors to review and merge it.

> - Each PR should include testing(Code testing section below).

Reviewer: ```suggestion
- Each PR should include testing (Code testing section below).
```

PR without tests won't be reviewed.
---
## Code Testing
Please use Pytest as your test framework.

* [Pytest](https://docs.pytest.org/en/7.2.x/getting-started.html)
- Please note that Pytest is already installed in your virtual environment.

---
## Review requirements 

> - When reviewing, confirm that the code accomplishes the requirements and the acceptance criteria,

Reviewer: Need to define, the requirements and acceptance criteria of what. GitHub issue? commit message? Something else?

If it's a GitHub issue, then you should define that each PR should be [tied to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue). If it's the commit message, then you need more elaborated criteria for the commit message content.

I think linking PRs to issues is easier to accomplish and more flexible.

Author: i think i will just delete this section, i dont find it much informative and i dont know how to elaborate it properly.

Reviewer: but that is how you decide whether the PR is doing what it's suppose to be doing or not. If you don't have that, then you can't tell whether the PR is actually doing what it's supposed to be doing.

Reviewer: You can say here that the PR needs to either be linked to a GitHub issue, or should specify in details what it's supposed to be achieving.

Author: added your notes, thanks!

# CONTRIBUTING
Contributing to the official buyTogether application
Welcome! We're so excited that you're considering contributing to the official GitHub page for buyTogether.
---
## Prerequisites

> - Fork the repo to make your changes without affecting the original project until you're ready to merge.

Reviewer 2: ready to propose your code changes*

The maintainers are the ones who should be ready to merge

- Create a working branch.

> - Set up a development environment to begin working on the project, all you need to do is 'vagrant up'

Reviewer: ```suggestion
- Set up a development environment to begin working on the project, all you need to do is 'vagrant up'.
```

> - Set up a development environment to begin working on the project, all you need to do is 'vagrant up'

Reviewer 2: Hey, I'm a new developer who wish to contribute to this project! I ran `vagrant up` like you suggested, and got `bash: vagrant: command not found`, any idea what the problem it? :)

Author: installing vagrant, right...✔

- Start with your changes!
---

> ## Commiting

Author: the section still exists, it is small, with a reference to outer guide, but still...
i think it answers on how to make a commit.

> Please use the guide below inorder to follow our commit message convention:

Reviewer: ```suggestion
Please use the guide below in order to follow our commit message convention:
```

> * [commit convention](https://cbea.ms/git-commit/#seven-rules)

Reviewer: I think this standard is a good fit. Just note that the commit message for this MR doesn't comply with it either. Specifically, [this point](https://cbea.ms/git-commit/#imperative). The commit message should be something like: `Modify contributing file`.

Reviewer: (to be clear, I think you should change this commit message rather than the standard)

Author: will change the title

---
## Pull Request
- When you're finished with the changes, create a pull request, also known as a PR.

> - If your PR depends on the merging of other PRs, indicate this on the commit message or on the comments.

Reviewer 2: don't indicate it in the comment section, indicate it on the commit message. 
also indicate it as such: `depends on #<number of PR>` so it'll be linked.

Specifically, if it contains unmerged commits of any other PRs,
indicate to the reviewers which commits are expected to be reviewed as part of your PR.
- Keep your PRs as small as possible.
Ideally, a single PR should include a single commit.
- The reviewer resolves the conversation to indicate that they are ok with the updated content.
- If you run into any merge issues, check how to solve them and resolve merge conflicts and other issues.

> - Changes made as part of the review process (i.e. fixes to the content of the original commit) ,

Reviewer: ```suggestion
- Changes made as part of the review process (i.e. fixes to the content of the original commit),
```

> - Changes made as part of the review process (i.e. fixes to the content of the original commit) ,

Reviewer 3: ```suggestion
- Changes made as part of the review process (i.e. fixes to the content of the original commit),
```

should be added to the original commit (e.g. using `commit --amend`), rather than creating a new commit. 

> - Please keep in mind that at least two team members of the group should "quality review" (i.e. NO LGTM)

Reviewer 2: `should provide a sufficient feedback` seems better IMO. there is a time and a place for LGTM, but it can't be the case in most of the time.

> and approve the pull request before asking the mentors to review and merge it.

Reviewer 2: s/mentors/maintainers.

Author: care to explain? s?

Reviewer: change `mentors` to `maintainers`

Reviewer: https://en.wikipedia.org/wiki/Sed#Substitution_command

> - Each PR should include testing (Code testing section below).

Reviewer 2: should include tests for the features it presents*

PR without tests won't be reviewed.
---
## Code Testing

> Please use Pytest as your test framework.

Reviewer 2: As you probably won't accept changes with different test frameworks, you don't have to be polite. `This project uses Pytest as its test framework`

Author: half British🤷‍♂️

* [Pytest](https://docs.pytest.org/en/7.2.x/getting-started.html)

> - Please note that Pytest is already installed in your virtual environment.

Reviewer 2: In this project's virtual environment*

---
## Review Requirements

> When reviewing, confirm that the code accomplishes the requirements and the acceptance criteria(Flake8, commit convention, etc.).

Reviewer: ```suggestion
When reviewing, confirm that the code accomplishes the requirements and the acceptance criteria (Flake8,
commit convention, etc.).
```

Reviewer: also, flake8 and commit conventions are not the acceptance criteria. The acceptance criteria is what this PR should accomplish (should be described on an issue or on the commit message - or both).

A GitHub issue: the code fixes the problem.
Adding a feature: elaborate in the commit the purpose of the code.

> Both cases should have accompanying unit tests that cover all the relevant edge cases.

Reviewer: Something is broken with the format of this paragraph. You can preview the file on GitHub or on an editor (online or on your machine), to see it.

---

## Commit your update
Please follow our project commit guide lines below:
- The first line is the title of the message,
it should be no longer than 63 characters.
- The second line should be left empty.
- The remaining consist of the body of the message,
they should be no longer than 72 characters each.
try to be descriptive as you can while keeping it simple.

- Example:
Adding an awesome feature.

Adding feature which:
1. makes cookies.
2. makes you get A+ in all courses.
3. create dank memes.

> in order to use each of the feature above just go to page ****,

Reviewer 4: No need for example here i think since all details has been written

Reviewer 3: Please notice that you commented on lines 30-36 (I think you meant to mark lines 29-30).

write the line "make_cookies()" where ever you want and walla, 

> cookies will appear.

Reviewer 3: For me, this example wasn't clear. 
First of all, I would write it inside a code block: `Example: this is a code block.`
And it is a little bit too long for me. The purpose here is to show how to write a good commit message.

Commit the changes once you are happy with them. 

---
## Review requirements 
- Before revewing the task, please acknowledge its requirements and acceptance criteria.

> 

Reviewer: ```suggestion
- When reviewing, confirm that the code accomplishes the requirements and the acceptance criteria, and that the accompanying unit tests cover all the relevant edge cases.

```

--- 
## Pull Request
- When you're finished with the changes, create a pull request, also known as a PR.

> - Please note that at least two team members of the group should quality review(NO LGTM) and approve the pull request before asking the mentors to review and merge.

Reviewer 4: NO LGTM ,should be betwen us but not in public i think you should delete it

Reviewer 2: Nothing should be kept private between you. You should consider this project as any open source project; if someone new would like to join the effort - this doc should let him know everything about how to contribute. CR process included.

> - The reviewer resolve the conversation to indicate that they are ok with the updated content.

Reviewer: ```suggestion
- The reviewer resolves the conversation to indicate that they are ok with the updated content.
```

> - If you run into any merge issues, check how to solve them and resolve merge conflicts and other issues.

Reviewer: Stuff which I find missing:

```
- If your PR depends on any other PRs being merged, indicate so on the commit message or on the comments. Specifically, if it contains unmerged commits of any other PRs, indicate to the reviewers which commits are expected to be reviewed as part of your PR.
- Keep your PRs as small as possible.
- Ideally, a single PR should include a single commit.
- A PR's commit, needs to represent a transition in the code that completes a task (e.g. a new function, along with its unit tests) .
- Changes made as part of the review process (i.e. fixes to the content of the original commit) , should be added to the original commit (e.g. using `commit --amend`), rather than creating a new commit. 
```
